### PR TITLE
Donations block: fix race condition when updating currency

### LINKED ISSUE #22838: Donations block: Saving race condition
### Impacted plugin

Jetpack

### Steps to Reproduce

 1. Add a donations block to a new post
 2. Change currency to another currency, e.g. Australian dollar
 3. Immediately press update post
 4. View the page
 5. Choose an amount
 6. Press Donate

### A clear and concise description of what you expected to happen.

The pop-up checkout should be in Australian dollars

### What actually happened

The pop-up checkout is in United States dollars

 Checkout screen:
image

### Other information

When I change currency a request to /wpcom/v2/memberships/products is made to update the currency of the linked plan. If this has not completed before I save the post then it will not update the currency used on the checkout screen. If I wait for that request to complete then it does update the currency.

Video of process
https://user-images.githubusercontent.com/93301/153851825-7222edf0-ad9f-463f-9d01-40a74c6eb219.mp4

### Operating System

_No response_

### OS Version

_No response_

### Browser

Firefox

### Browser Version(s)

_No response_

### PR DESCRIPTION
**Pull request by rcrdortiz**
Fixes #22838
When we change the currency of a donations block we can save the post before we've created the plan for the given currency, this causes the block to display the currency set by the user but we don't have a subscription plan for it so the previous subscription plan is shown when paying through the payment form.

#### Changes proposed in this Pull Request:

* Now the update button of the post is blocked until we complete the operation of creating the subscription plan for the selected currency.

#### Does this pull request change what data or activity we track or use?
No

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Add a donations block to a new post and publish the post.
* Change currency to another currency, e.g. Australian dollar
* Update the post as fast as you can (press the update button when available)
* View the page
* Choose an amount
* Press Donate
* The currency should match the currency of the block.

#### Test evidence:
https://user-images.githubusercontent.com/1989914/188640556-157e2667-25d9-476c-b56f-7e53dbe4c62c.mp4

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/donations/edit.js [hunk 1]
```diff
@@ -66,16 +69,19 @@ const Edit = props => {
...
+			setProducts( filterProducts( defaultProducts ) );
+			unlockPostSaving( 'donations' );
 		}, apiError );
 	};
 
 	useEffect( () => {
+		lockPostSaving( 'donations' );
 		const updateData = () => fetchStatus( 'donation' ).then( mapStatusToState, apiError );
 		updateData();
 	}, [ currency ] );
```
Comment (mmtr) [inline projects/plugins/jetpack/extensions/blocks/donations/edit.js]: We need to update the dependencies of this effect:

```
ESLint: React Hook useEffect has a missing dependency: 'lockPostSaving'. Either include it or remove the dependency array.(react-hooks/exhaustive-deps)
```

**Review by mmtr — COMMENTED** [review]
Works as advertised! Just left a minor comment.

File: projects/plugins/jetpack/extensions/blocks/donations/edit.js (hunk 1 above)
Comment (rcrdortiz) [reply projects/plugins/jetpack/extensions/blocks/donations/edit.js]: Updated

**Review by mmtr — APPROVED** [review]

**Comment by rcrdortiz** [comment]
r252170-wpcom

**Comment by wynwin** [comment]
6186301-zen
@rcrdortiz We got a similar report of this above, so wanted to check on the latest here? Looks like it has been merged, but then there is a comment showing not verified? I want to make sure I understand where this one currently stands.

**Comment by wynwin** [comment]
> 6186301-zen @rcrdortiz We got a similar report of this above, so wanted to check on the latest here? Looks like it has been merged, but then there is a comment showing not verified? I want to make sure I understand where this one currently stands.

@rcrdortiz Tested again and this now seems good to go? Disregard. Thank you!

**Pull request merged**